content/actions/automating-builds-and-tests/building-and-testing-python.md

```diff
@@ -410,24 +410,45 @@ on:
     types: [published]
 
 jobs:
-  deploy:
+  release-build:
     runs-on: ubuntu-latest
+
     steps:
       - uses: {% data reusables.actions.action-checkout %}
-      - name: Set up Python
-        uses: {% data reusables.actions.action-setup-python %}
+
+      - uses: {% data reusables.actions.action-setup-python %}
         with:
-          python-version: '3.x'
-      - name: Install dependencies
+          python-version: "3.x"
+
+      - name: build release distributions
         run: |
-          python -m pip install --upgrade pip
-          pip install build
-      - name: Build package
-        run: python -m build
-      - name: Publish package
-        uses: pypa/gh-action-pypi-publish@release/v1
+          # NOTE: put your own distribution build steps here.
+          python -m pip install build
+          python -m build
+
+      - name: upload windows dists
+        uses: {% data reusables.actions.action-upload-artifact %}
         with:
-          password: {% raw %}${{ secrets.PYPI_API_TOKEN }}{% endraw %}
+          name: release-dists
+          path: dist/
+
+  pypi-publish:
+    runs-on: ubuntu-latest
+    needs:
+      - release-build
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Retrieve release distributions
+        uses: {% data reusables.actions.action-download-artifact %}
+        with:
+          name: release-dists
+          path: dist/
+
+      - name: Publish release distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
 ```
 
-For more information about the starter workflow, see [`python-publish`](https://github.com/actions/starter-workflows/blob/main/ci/python-publish.yml).
+For more information about this workflow, including the PyPI settings
+needed, see [AUTOTITLE](/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-pypi).
```

Comment:
NB: This removes the link to the starter workflow, since it's also currently out-of-date. I'll send a PR updating it as well.